### PR TITLE
Feature/order list redesign changes master

### LINKED ISSUE #1398: Order List redesign - Master checklist
This is the master epic for the changes required to accomplish the order list improvements.

#### Changes in FluxC:
- [x] [#1372](https://github.com/wordpress-mobile/WordPress-FluxC-Android/issues/1372) - Cache the total order status count from the order status total api response (PR: [#1376](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1376), [#1389](https://github.com/wordpress-mobile/WordPress-FluxC-Android/issues/1389))

#### UI Changes:
- [x] #1399 - Add `TabLayout` to the order list UI: `Processing` and `All Orders` (PR: #1404 )
   - [x] Anchor the `TabLayout` when the view is scrolled.
   - [x] Remember the `Tab` selected and default to that tab. Default is `Processing`.
- [ ] #1411 - Handle empty state scenarios in order list (PR: #1430)
- [x] #1400 - Enhance search support to the order list UI (PR: #1428)
   - [x] Remove the filter option from the toolbar menu
   - [x] Add new search UI that displays list of order statuses along with their count.
   - [x] Clicking on an order status from the above UI will display the list of orders for that status.

 .

### PR DESCRIPTION
**Pull request by anitaa1990**
Fixes #1398 . This is the master branch for the order list redesign feature. All commits below have been reviewed already ~~but this [step 3 PR](https://github.com/woocommerce/woocommerce-android/pull/1430) should be merged first before merging this PR.~~

### Changes:
- Added a new class `OrderListView` that handles the order list related logic such as populating the `RecyclerView`, displaying empty view.
- Added `TabLayout` with `Processing` and `All Orders` tab and anchor the `TabLayout` to the top of the view when scrolled.
- Added a new class `OrderStatusListView` that displays the order status list UI when search icon is selected.
- Removed the filter option from the toolbar menu.

### Screenshots:
 . 

#### Search Behaviour
 . 

#### Empty View: No orders for a site:
 . 

#### Empty View: No unfulfilled orders for a site:
 .  

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/OrderListPresenter.kt [hunk 1]
```diff
@@ -84,6 +84,12 @@ class OrderListPresenter @Inject constructor(
             orderView?.showLoading(true)
             val payload = FetchOrdersPayload(selectedSite.get(), filterByStatus)
             dispatcher.dispatch(WCOrderActionBuilder.newFetchOrdersAction(payload))
+        } else if (!networkStatus.isConnected()) {
+            orderView?.let { order ->
+                order.showLoadOrdersError()
```
Comment (nbradbury) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/OrderListPresenter.kt]: Shouldn't this call `showNoConnectionError()` instead?

File: WooCommerce/src/main/res/values/colors.xml [hunk 2]
```diff
@@ -45,6 +45,7 @@
     <color name="default_text">@color/wc_grey_dark</color>
     <color name="default_text_title">@color/grey_dark_highlight</color>
     <color name="default_text_title_disabled">@color/wc_grey_17</color>
+    <color name="default_search_hint_text">#c5adc3</color>
```
Comment (nbradbury) [inline WooCommerce/src/main/res/values/colors.xml]: This isn't an issue with this PR, but just so it's addressed can you update `drawable/searchview_cursor.xml` to use this? Right now it has the same color hard-coded.

Comment (anitaa1990) [reply WooCommerce/src/main/res/values/colors.xml]: 👌 Fixed in 04822e9

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/OrderListPresenter.kt (hunk 1 above)
Comment (anitaa1990) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/OrderListPresenter.kt]: Good point! Fixed in 92c322d

**Comment by anitaa1990** [comment]
Thanks for the quick review @nbradbury! Fixed all the comments and PR ready for another review 👍

**Comment by nbradbury** [comment]
@anitaa1990 Changes look good, but the tests are now failing.

**Comment by anitaa1990** [comment]
Shoot! I didn't notice that CircleCI failed. Fixed the test in fa37f12 👍

**Pull request merged**